### PR TITLE
Fix bug with displayed sort option on Sort By select box

### PR DESCRIPTION
**Pull request by karlbaker02**
This PR fixes the Sort By select box to display the correct sort option when the page is refreshed.

Trello card: https://trello.com/c/zdcfNCoZ

Solo: @karlbaker02

### CODE REVIEW

File: app/presenters/finder_presenter.rb [hunk 1]
```diff
@@ -136,10 +136,7 @@ def sort_options
 
     disabled_option = keywords.blank? ? relevance_sort_option : ''
 
-    selected_option = if values['order']
-                        sort.detect { |option| option['name'].parameterize == values['order'] }
-                      end
-
+    selected_option = values['order'] unless values['order'].nil? && sort.detect { |option| option['name'].parameterize == values['order'] }.nil?
```
Comment (andrewgarner) [inline app/presenters/finder_presenter.rb]: We could structure the code like so to reduce the line length and improve readability?

```ruby
def sort_values
  sort
    .map { |option| option['name'].parameterize }
    .compact
end

selected_option = values['order'] if sort_values.include?(values['order'])
selected_option ||= default_sort_option_value
```

Comment (koetsier) [reply app/presenters/finder_presenter.rb]: can we add a test for this?

Comment (karlbaker02) [reply app/presenters/finder_presenter.rb]: @andrewgarner Your suggestion definitely improves on line length and readability, at the expense of performing the computation of `sort_values` regardless of whether of `values['order']` is present. We can definitely improve on the readability though.

@koetsier I'm sure we could

Comment (karlbaker02) [reply app/presenters/finder_presenter.rb]: Tests added @koetsier

File: spec/presenters/finder_presenter_spec.rb [hunk 2]
```diff
@@ -68,4 +80,77 @@
...
+    end
+
+    it "returns sort options when sort is present" do
+      expected_options = "<option value=\"most-viewed\">Most viewed</option>\n<option value=\"updated-newest\">Updated (newest)</option>"
+
+      expect(presenter_with_sort.sort_options).to eql(expected_options)
+    end
+
+    it "returns sort options without relevance when keywords is not present" do
+      expected_options = "<option value=\"most-viewed\">Most viewed</option>\n<option value=\"updated-newest\">Updated (newest)</option>"
```
Comment (koetsier) [inline spec/presenters/finder_presenter_spec.rb]: this test is identical to the one starting line 89

File: app/presenters/finder_presenter.rb (hunk 1 above)
Comment (koetsier) [reply app/presenters/finder_presenter.rb]: I'm not sure this is right:

```selected_option = values['order'] unless values['order'].nil? && sort.detect { |option| option['name'].parameterize == values['order'] }.nil?```

is the same as

```selected_option = values['order'] if !values['order'].nil? || !sort.detect { |option| option['name'].parameterize == values['order'] }.nil?```

so this implies that in the second part ```values['order']``` can be set to nil and so the second part could be simplified to ```sort.any?{| |option| option['name'].nil?}

File: spec/presenters/finder_presenter_spec.rb (hunk 2 above)
Comment (karlbaker02) [reply spec/presenters/finder_presenter_spec.rb]: It's close, but not the same. The test on line 85 uses `presenter`, which has `nil` sort options. This test on line 95 uses `presenter_with_sort`, which has sort options (including relevance).

File: app/presenters/finder_presenter.rb (hunk 1 above)
Comment (karlbaker02) [reply app/presenters/finder_presenter.rb]: You're right, not the same.

This should actually be `unless ... || ...`. Will update.

File: app/presenters/finder_presenter.rb [hunk 3]
```diff
@@ -134,15 +134,12 @@ def sort_options
...
 
-    disabled_option = keywords.blank? ? relevance_sort_option : ''
-
-    selected_option = if values['order']
-                        sort.detect { |option| option['name'].parameterize == values['order'] }
-                      end
+    disabled_option = keywords.blank? ? relevance_sort_option_value : ''
 
+    selected_option = values['order'] unless values['order'].nil? || sort.detect { |option| option['name'].parameterize == values['order'] }.nil?
     selected_option ||= default_sort_option_value
```
Comment (koetsier) [inline app/presenters/finder_presenter.rb]: what about:

```
    selected_option = if values['order'].present? && sort.any? { |option| option['name'].parameterize == values['order'] }
                        values['order']
                      else
                        default_sort_option_value
                      end
```

Comment (karlbaker02) [reply app/presenters/finder_presenter.rb]: Updated @koetsier

**Pull request merged**